.github/workflows/update-deps.yml

```diff
@@ -0,0 +1,71 @@
+name: Update NPM dependencies
+
+on:
+  schedule:
+  - cron: "0 5 * * 1"
+
+jobs:
+  update:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        deps: [production, dev-only]
+
+    steps:
+      - name: Fetch changes
+        # https://github.com/marketplace/actions/checkout
+        uses: actions/checkout@v2
+        with:
+          ref: master
+
+      - name: Setup node
+        # https://github.com/marketplace/actions/setup-node-js-environment
+        uses: actions/setup-node@v1
+        with:
+          node-version: '12'
+
+      - name: Is update possible
+        run: |
+          echo "BRANCH_NAME=deps-update_${{ matrix.deps }}" >> $GITHUB_ENV
+          if [[ $(git ls-remote --heads | grep deps-update_${{ matrix.deps }} | wc -c) -ne 0 ]]; then
+            echo "BRANCH_NAME=0" >> $GITHUB_ENV
+          fi
+
+      - name: Update dependencies
+        if: env.BRANCH_NAME != 0
+        run: |
+          npm i
+          npm install -g npm-check
+          git checkout -b ${{ env.BRANCH_NAME }}
+          npm-check -y --${{ matrix.deps }}
+
+      - name: Add changes
+        if: env.BRANCH_NAME != 0
+        run: |
+          if [[ $(git diff origin/master | wc -c) -ne 0 ]]; then
+            git config --local user.email "ckeditor-bot@cksource.com"
+            git config --local user.name "CKEditor Bot"
+            git add package*.json
+            git commit -m "Update NPM ${{ matrix.deps }} dependencies."
+            echo "HAS_CHANGES=1" >> $GITHUB_ENV
+          fi
```

Comment:
It would be good to move variables which could change in the future as environment variables, so:
`user.email`
`user.name`
`origin/master` - > especially this one, not sure if it's likely that email and name of the bot would change.